spritecontrol.h

```diff
@@ -89,10 +90,26 @@ class SpriteChoreographer {
 	bool should_change_pattern();
 	void update_player();
 
+	inline static std::vector<PatternName> m_enabled_patterns = { };
+
 	Sprites *m_sprites;
 	Context *m_ctx;
 	PatternName m_pattern;
 	std::vector<PatternPlayer *> m_players;
 	PatternPlayer *m_current_player;
 };
 
+class PatternRepository {
+public:
+	static std::vector<PatternName> load_disabled_patterns();
+	static std::vector<PatternName> get_enabled_patterns(const std::vector<PatternName> &disabled_patterns);
+	inline static std::vector<PatternName> load_enabled_patterns() { return get_enabled_patterns(load_disabled_patterns()); }
```

Comment:
Even though it's a really simple function I would move this to a .cpp file like the rest of them. It just makes it easier to look for and find, I got lost since I didn't realize you'd defined it inline here.